server/controllers/userController.js

```diff
@@ -0,0 +1,88 @@
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const passport = require("passport");
+const { body, validationResult } = require("express-validator");
+const User = require("../models/User");
+
+exports.create = [
+  body("username")
+    .notEmpty()
+    .withMessage("Username is required")
+    .custom(async (value) => {
+      const user = await User.findOne({ username: value });
+      return user
+        ? Promise.reject("Username already in use.")
+        : Promise.resolve(true);
+    })
+    .trim(),
+  body("email")
+    .notEmpty()
+    .withMessage("Email is required")
+    .isEmail()
+    .withMessage("Must be valid email address")
+    .custom(async (value) => {
+      const user = await User.findOne({ email: value });
+      return user
+        ? Promise.reject("Email already in use.")
+        : Promise.resolve(true);
+    })
+    .trim()
+    .normalizeEmail(),
+  body("password")
+    .isLength({ min: 8, max: 60 })
+    .withMessage("Passwords must be at least 8 characters long.")
+    .trim(),
```

Comment:
We might want to define validation in a separate file as middleware to avoid cluttering up the controller.

Reply:
Right on! Now in their own file and controllers looking much cleaner.